src/internal.c

```diff
@@ -26128,25 +26129,38 @@ int SetCipherList(WOLFSSL_CTX* ctx, Suites* suites, const char* list)
     const int suiteSz       = GetCipherNamesSize();
     const char* next        = list;
 
-    if (suites == NULL || list == NULL) {
+    ProtocolVersion version;
+    int privateKeySz = 0;
+    byte side;
+#ifdef HAVE_ANON
+    byte haveAnon = 0;
+#endif
+
+    if (suites == NULL || list == NULL || (ctx == NULL && ssl == NULL)) {
         WOLFSSL_MSG("SetCipherList parameter error");
         return 0;
     }
 
+    version = ctx != NULL ? ctx->method->version : ssl->version;
```

Comment:
The only difference I can see when using ctx or ssl is that the version, privateKeySz and side are coming from ctx or ssl.
Extract the code that does the work into a new function and have 'haveAnon' be returned through a parameter.
Leave the SetCipherList API as it was and have it call the new function.
Add a new function SetCipherListEx that takes the ctx and ssl and uses the new function.